repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/UserRepositoryTestImp.scala

```diff
@@ -39,7 +39,8 @@ class UserRepositoryTestImp extends UserRepository {
 
   def updateUser(updatedUser: User): Future[Int] = update(updatedUser)
 
-  def updatePassword(updatedUser: User): Future[Int] = update(updatedUser)
+  def updatePassword(userId: UserId, hash: String, salt: String): Future[Int] =
+    users.get(userId).fold(Future.successful(0))(u => update(u.copy(passwordHash = hash, passwordSalt = salt)))
```

Comment:
If i got it right, we check whether there is a user under provided userId, and if there is, we check the same condition again in the update method. May be in that case it will be better to delete private update method instead of reusing it?

Reply:
It bothers me a bit as well, but if I remove `update` method and inline it's code in `updateUser` and `updatePassword` methods, then we will have to copy-paste some code. IMO double check with fairly readable code is better than code duplication.

Reply:
Got it, thanks!